cassandra/cassandra_test.go

```diff
@@ -0,0 +1,279 @@
+package cassandra
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gocql/gocql"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	cassandracontainer "github.com/testcontainers/testcontainers-go/modules/cassandra"
+)
+
+const (
+	// cassandraImage is the default image used for running cassandra in tests.
+	cassandraImage              = "cassandra:latest"
+	cassandraImageEnvVar string = "TEST_CASSANDRA_IMAGE"
+	cassandraPort               = "9042/tcp"
+)
+
+// newTestStore creates a Cassandra container using the official module
+func newTestStore(t testing.TB) *Storage {
+	t.Helper()
+
+	img := cassandraImage
+	if imgFromEnv := os.Getenv(cassandraImageEnvVar); imgFromEnv != "" {
+		img = imgFromEnv
+	}
+
+	ctx := context.Background()
+
+	c, err := cassandracontainer.Run(ctx, img)
+	testcontainers.CleanupContainer(t, c)
+	require.NoError(t, err)
+
+	host, err := c.Host(ctx)
+	require.NoError(t, err)
+
+	port, err := c.MappedPort(ctx, cassandraPort)
+	require.NoError(t, err)
+
+	store, err := New(Config{
+		Hosts:       []string{fmt.Sprintf("%s:%d", host, port.Int())},
```

Comment:
suggestion: let's use testcontainers API for cassanda
```suggestion
	connectionHost, err := c.ConnectionHost(ctx)
	require.NoError(t, err)

	store, err := New(Config{
		Hosts:       []string{connectionHost},
```